HealthMonitoring.Management.Core/Registers/EndpointRegistry.cs

```diff
@@ -81,6 +83,7 @@ public void UpdateHealth(Guid endpointId, EndpointHealth health)
                 return;
             endpoint.UpdateHealth(health);
             _statsRepository.InsertEndpointStatistics(endpointId, health);
+            _metricsForwarderCoordinator.HandleMetricsForwarding(endpoint);

```

Comment:
So as I written above I would ensure here that we are forwarding health instance not endpoint.health, that could mutate in meantime